Documentation/devel/distribution.md

```diff
@@ -0,0 +1,145 @@
+`cimd:aci-archive:v=0:https%3A%2F%2Fexample.com%2Fapp.aci`
+
+**Docker**
+This is a Name based distribution
+
+Docker defines a distribution using a docker registry
+The format is the same as the docker image string format (man docker-pull) with the "docker" scheme:
+`cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG|@DIGEST]`
+Examples:
+`cimd:docker:v=0:busybox`
+`cimd:docker:v=0:busybox:latest`
+`cimd:docker:v=0:registry-1.docker.io/library/busybox@sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6`
+
+### Future distributions
+
+**OCI Image distributions**
+This is a Name based distribution
+
+Now a docker distribution can also provide oci image but in future oci image will define one or more own kind of distribution starting from an image name (and additional tags, labels)
+
+**OCI Image layout**
+This is a Location based distribution
+
+It can fetch an image starting from a [OCI image layout](https://github.com/opencontainers/image-spec/blob/master/image-layout.md) format. The location can point to a single file archive, to a local/remote directory based layout or other kind of locations
+
+Probably it will be the sub-distribution used by the OCI image distribution described above) (like ACIArchive is the sub-distribution for the appc distribution).
+
+`cimd:oci-image-layout:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile?ref=refname`
+`cimd:oci-image-layout:v=0:https%3A%2F%2Fdir%2F?ref=refname`
+
+Since the OCI image layout can provide multiple images selectable by a ref there's the need to specify which ref to use in the archive distribution URI (see the above ref query parameter). Since distribution just covers one image it's not possible to import all the refs with just a distribution URI.
+
+TODO(sgotti). Define if oci-image-layout should internally handle both archive and directory based layouts or use two different distributions or a query parameter the explicitly define the layout (to avoid guessing if the url points to a single file or to a directory).
+
+**Note** Considering [this OCI image spec README section](https://github.com/opencontainers/image-spec#running-an-oci-image), probably the final distribution format will be similar to the app distribution. So there's a need to distinguish their User Friendly string (prepending an appc: or oci: ?).
+
+To sum it up:
+
+| Distribution   | Type     | URI Format                                                                | Sub-distribution |
+|----------------|----------|---------------------------------------------------------------------------|------------------|
+| Appc           | Name     | `cimd:appc:v=0:name?label01=....&label02=...`                             | ACIArchive       |
+| Docker         | Name     | `cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG&#124;@DIGEST]` | <none>           |
+| ACIArchive     | Location | `cimd:aci-archive:v=0:ArchiveURL?query...`                                |                  |
```

Comment:
I'm not sure I completely understand your question. I'll add more contents but please let me know if there's something not clear or if you don't agree on something.

* The distribution is only defined by its URI.
* The distribution Type is only a logical distinction but I think it's very important to clarify this distinction and its role since it'll affect the logic of the various distribution related implementations (like #2964 #3071). I tried to explain it [here](https://github.com/coreos/rkt/pull/2953/files/50c7a2f3072f67bb6254e1a09270e6e52ac168d9#diff-28042cdb928d9fb5227b3ced90a10356R13).

I'll try to explain my reasoning behind this:
* Name based distribution should be the primary way to fetch/reference an image.
 * for example appc ACI dependencies can be resolved only by a name based distribution (appc).
* Location based distribution should not be directly used because they map a fixed location to an image but they exists and are needed for different reason:
 * rkt provides a way to fetch and run an image by a location (the current file or http uri are an aci-archive distribution)
 * Some name based distribution will naturally resolve to a location based distribution:
  * appc distribution will do discovery and the discovered url is a location based distribution (aci-archive)
  * the same will probably apply for OCI image (from a name to an oci-image-layout location distribution)
 * in a reference based store logic (see the notes in https://github.com/coreos/rkt/pull/3071/commits/23313af1c3dac2fb24fe41f9a7c5eaca573e45dd#diff-146353160c5a20311afdb1ccda76b178R299) it could be useful to let the user map a location based distribution to a name based one just because a name based is needed for appc distribution but I'd like to "import" an image directly by a URL. Instead it doesn't makes sense to map a name to a location or a location to a location distribution.

I intentionally avoided the term "discovery" because:
* it's just a way to implement a name based distribution
* I don't consider a location distribution related with discovery, you already provide the final location where the image is contained (aci-archive, oci-image-layout).